src/lib/ngsi/Metadata.cpp

```diff
@@ -428,8 +429,7 @@ std::string Metadata::toStringValue(void) const
     }
     else // regular number
     {
-      snprintf(buffer, sizeof(buffer), "%f", numberValue);
-      return std::string(buffer);
+      return toString(numberValue);
```

Comment:
The fix in this function is the one mentioned in CNR as:

> Fix: removed spurious decimals in metadata numbers in NGSIv1 rendering (a follow up of #2176)

We need this otherwise we would get thinks like this in NGSIv1 notifications:

```
                    {
                        "metadatas": [
                            {
                                "name": "previousValue",
                                "type": "Number",
                                "value": 20.0
                            }
                        ],
                        "name": "B",
                        "type": "Number",
                        "value": 20
                    }
```

which is a bit weird.

NTC (just informative).